<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>677</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>10</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>장치 선택</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>10</y>
      <width>58</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>장치 설정</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>46</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>모듈:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cbo_module">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>40</y>
      <width>111</width>
      <height>31</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Zigbee HA</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Zigbee 3.0</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>BLE</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>UART</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>40</y>
      <width>41</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>이름:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>70</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>uuid:</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>100</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>address:</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_device_name">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>40</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_device_uuid">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>70</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_device_addr">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>100</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_device_ep">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>130</y>
      <width>201</width>
      <height>21</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>130</y>
      <width>31</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>ep:</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>511</width>
      <height>401</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <property name="iconSize">
     <size>
      <width>16</width>
      <height>16</height>
     </size>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>single</string>
     </attribute>
     <widget class="QTabWidget" name="tab_single">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>20</y>
        <width>491</width>
        <height>171</height>
       </rect>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>connect</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>on/off</string>
       </attribute>
       <widget class="QRadioButton" name="rdo_on">
        <property name="geometry">
         <rect>
          <x>50</x>
          <y>100</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>on</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="rdo_off">
        <property name="geometry">
         <rect>
          <x>120</x>
          <y>100</y>
          <width>41</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>off</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="rdo_toggle">
        <property name="geometry">
         <rect>
          <x>180</x>
          <y>100</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>toggle</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_20">
        <property name="geometry">
         <rect>
          <x>110</x>
          <y>20</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>입력값 설정</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_21">
        <property name="geometry">
         <rect>
          <x>280</x>
          <y>20</y>
          <width>51</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>반복 횟수 </string>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_onoff">
        <property name="geometry">
         <rect>
          <x>280</x>
          <y>50</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>300</number>
        </property>
       </widget>
       <widget class="QComboBox" name="cbo_input_onoff">
        <property name="geometry">
         <rect>
          <x>40</x>
          <y>40</y>
          <width>221</width>
          <height>31</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>직접 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>정상 범주 내 랜덤값 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>비정상 범주 내 랜덤값 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>모든 범주의 랜던값 입력</string>
         </property>
        </item>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>color</string>
       </attribute>
       <widget class="QComboBox" name="cbo_input_color">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>60</y>
          <width>221</width>
          <height>31</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>직접 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>정상 범주 내 랜덤값 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>비정상 범주 내 랜덤값 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>모든 범주의 랜던값 입력</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_24">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>30</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>입력값 설정</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit_color">
        <property name="geometry">
         <rect>
          <x>260</x>
          <y>60</y>
          <width>113</width>
          <height>21</height>
         </rect>
        </property>
        <property name="placeholderText">
         <string>0x00 직접 입력</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_color">
        <property name="geometry">
         <rect>
          <x>390</x>
          <y>60</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>300</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_22">
        <property name="geometry">
         <rect>
          <x>390</x>
          <y>20</y>
          <width>51</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>반복 횟수 </string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_6">
       <attribute name="title">
        <string>level</string>
       </attribute>
       <widget class="QComboBox" name="cbo_input_level">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>50</y>
          <width>221</width>
          <height>31</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>직접 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>정상 범주 내 랜덤값 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>비정상 범주 내 랜덤값 입력</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>모든 범주의 랜던값 입력</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_25">
        <property name="geometry">
         <rect>
          <x>70</x>
          <y>20</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>입력값 설정</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit_level">
        <property name="geometry">
         <rect>
          <x>240</x>
          <y>60</y>
          <width>111</width>
          <height>21</height>
         </rect>
        </property>
        <property name="inputMask">
         <string/>
        </property>
        <property name="placeholderText">
         <string>0x00 직접 입력</string>
        </property>
       </widget>
       <widget class="QSpinBox" name="spinBox_level">
        <property name="geometry">
         <rect>
          <x>390</x>
          <y>60</y>
          <width>61</width>
          <height>21</height>
         </rect>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>300</number>
        </property>
       </widget>
       <widget class="QLabel" name="label_23">
        <property name="geometry">
         <rect>
          <x>390</x>
          <y>30</y>
          <width>51</width>
          <height>21</height>
         </rect>
        </property>
        <property name="text">
         <string>반복 횟수 </string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="tab_7">
       <attribute name="title">
        <string>disconnect</string>
       </attribute>
      </widget>
     </widget>
     <widget class="QPushButton" name="btn_insert">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>190</y>
        <width>112</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>추가</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>routine</string>
     </attribute>
     <widget class="QSpinBox" name="spinBox_routine">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>330</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>300</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_onoff_routine">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>140</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>300</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_color_routine">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>210</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="maximum">
       <number>300</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_level_routine">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>260</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="maximum">
       <number>300</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>330</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>루틴의 반복 횟수 </string>
      </property>
     </widget>
     <widget class="QLabel" name="label_26">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>50</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>입력값 설정</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_27">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>40</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>반복 횟수 </string>
      </property>
     </widget>
     <widget class="QComboBox" name="cbo_routine">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>10</y>
        <width>281</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>connect-on/off-color-level-disconnect</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>connect-on/off-level-color-disconnect</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>connect-color-on/off-level-disconnect</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>connect-level-on/off-color-disconnect</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>connect-color-level-on/off-disconnect</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>connect-level-color-on/off-disconnect</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>20</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>루틴 선택 </string>
      </property>
     </widget>
     <widget class="QLabel" name="label_29">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>on/off</string>
      </property>
     </widget>
     <widget class="QComboBox" name="cbo_input_onoff_routine">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>130</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>직접 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>정상 범주 내 랜덤값 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>비정상 범주 내 랜덤값 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>모든 범주의 랜던값 입력</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_30">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>260</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>level</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_31">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>210</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>color</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="rdo_on_routine">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>160</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>on</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="rdo_off_routine">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>160</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>off</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="rdo_toggle_routine">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>160</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>toggle</string>
      </property>
     </widget>
     <widget class="QComboBox" name="cbo_input_level_routine">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>260</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>직접 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>정상 범주 내 랜덤값 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>비정상 범주 내 랜덤값 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>모든 범주의 랜던값 입력</string>
       </property>
      </item>
     </widget>
     <widget class="QComboBox" name="cbo_input_color_routine">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>210</y>
        <width>221</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>직접 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>정상 범주 내 랜덤값 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>비정상 범주 내 랜덤값 입력</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>모든 범주의 랜던값 입력</string>
       </property>
      </item>
     </widget>
     <widget class="QLineEdit" name="lineEdit_level_routine">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>260</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="inputMask">
       <string/>
      </property>
      <property name="placeholderText">
       <string>0x00 직접 입력</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_color_routine">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>210</y>
        <width>113</width>
        <height>21</height>
       </rect>
      </property>
      <property name="placeholderText">
       <string>0x00 직접 입력</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btn_insert_routine">
      <property name="geometry">
       <rect>
        <x>290</x>
        <y>330</y>
        <width>112</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>추가</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>10</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>결과 / 진행</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_13">
    <property name="geometry">
     <rect>
      <x>950</x>
      <y>220</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>최종 결과</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_14">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>210</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>생성된 command</string>
    </property>
   </widget>
   <widget class="QListView" name="listView_command">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>280</y>
      <width>271</width>
      <height>291</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_more">
    <property name="geometry">
     <rect>
      <x>860</x>
      <y>580</y>
      <width>112</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>more</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_save">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>580</y>
      <width>112</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>save</string>
    </property>
   </widget>
   <widget class="QListView" name="listView_process">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>40</y>
      <width>331</width>
      <height>151</height>
     </rect>
    </property>
   </widget>
   <widget class="QListView" name="listView_result">
    <property name="geometry">
     <rect>
      <x>870</x>
      <y>250</y>
      <width>201</width>
      <height>321</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_start">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>580</y>
      <width>112</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>실험시작 </string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_up">
    <property name="geometry">
     <rect>
      <x>560</x>
      <y>240</y>
      <width>112</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>위로</string>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_down">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>240</y>
      <width>112</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>아래로</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1077</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <widget class="QMenu" name="menuImport_json">
     <property name="title">
      <string>Import json</string>
     </property>
     <addaction name="action_import_device"/>
     <addaction name="action_import_command"/>
    </widget>
    <widget class="QMenu" name="menuexport_json">
     <property name="title">
      <string>export json</string>
     </property>
     <addaction name="action_export_device"/>
     <addaction name="action_export_command"/>
    </widget>
    <addaction name="menuImport_json"/>
    <addaction name="menuexport_json"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="action_import_device">
   <property name="text">
    <string>import device</string>
   </property>
  </action>
  <action name="action_import_command">
   <property name="text">
    <string>import command</string>
   </property>
  </action>
  <action name="action_export_device">
   <property name="text">
    <string>export device</string>
   </property>
  </action>
  <action name="action_export_command">
   <property name="text">
    <string>export command</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
 <slots>
  <slot>slot_test()</slot>
 </slots>
</ui>
